sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+BlockBlobURL bu = new BlockBlobURL(parts.toURL(), StorageURL.createPipeline(
+    new AnonymousCredentials(), new PipelineOptions()))
+```
+
+In v12, SAS tokens can be managed just like any other credential.
+```java
+// Initialize the credentials
+SASTokenCredential credential = SASTokenCredential.fromSASTokenString("${sasToken}");
+// How service object take in the credentials.
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net")
+    .credential(credential)
+    .buildBlobServiceClient();
+```
+
+However, if you already have the full URL with SAS token attached, you can use that too:
+```java
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net/?${sasToken}")
+    .buildBlobServiceClient();
+```
+
+## Provide side by side sync-over-async APIs
+Each type (Container, Blob, etc.) has both a sync and async client. It is easy and intuitive to use whichever one is preferable in a given context without having to bother about clutter from the other.
+Suppose the Container Client been initialized as `${containerClient}`. Please refer the [Build Client](#build-client) section.
+
+* Example of using Async API in V12. 
+```java
+Mono<BlobProperties> response = containerClient.getProperties();
+```
+
+* Example of using Sync API in V12
+```java
+BlobProperties blobProperties = containerClient.getProperties();
+```
+
+## Async APIs Comparison
+In v10/v11, every RESTful API returned a reactive response of type `Single`
+(see RxJava documentation for more details), which emits one and only one result.
+These returned object representations of the single HTTP response that came back.
+
+Single result API in v10/v11:
+```java
+serviceURL.getProperties(Context.NONE).subscribe(
+    // print the selected service property on success
+    response -> System.out.println(response.body().defaultServiceVersion()),
+    // handle error
+    error -> System.out.println(error.getMessage()));
+```
+
+Listing API in v10/v11:
+```java
+serviceURL.listContainersSegment(null, null, Context.NONE).subscribe(
+    // print the list; grab continuation token for making another call to get next list segment
+    response -> {
+        System.out.println("Page token to fetch more containers:" + response.body().nextMarker());
+        response.body().containerItems().stream().forEach(
+            containerItem -> System.out.println("Container name: " + containerItem.name()));
+    },
+    // handle error
+    error -> System.out.println(error.getMessage()));
+```
+
+In v12, Reactor provides an equivalent class `Mono` (0-1 response) and provide our own extension of Reactor's `Flux` (0-many responses) for listing operations.
+Additionally, classes returned from APIs are the "core information" of the response, rather than the HTTP response exactly.
+Full access to all fields of the HTTP response is available, but beyond the scope of this document.
```

Comment:
Can you at least add "... is available by using the methods having the suffix -withResponse, but examples of that usage are beyond the scope...". That way they at least know where to look even if we don't give an example.